src/proto/mod.rs

```diff
@@ -28,16 +33,17 @@ pub use batch::{Batch, BatchBuilder, BatchHandle, BatchStatus, CallbackState};
 
 /// A stream that can be re-established after failing.
 #[async_trait::async_trait]
-pub trait Reconnect: Sized {
+pub trait Reconnect {
```

Comment:
I think you'll want to add a blanket impl of this trait for `Box`:
```rust
impl<S> Reconnect for Box<S> where S: Reconnect
```

Reply:
done